### PR TITLE
fix: Use segmentInfo.trackInfo to get segment information.

### PR DESCRIPTION
**Pull request by brandonocasey**
## Description
This prevents an issue shown via the test that I added. Mainly that a playlist change that happens before we append data from a `pendingSegment_` will cause reset `this.currentMediaInfo_` to be null. Thus causing `handleData_`'s call to `hasEnoughInfoToAppend_`  to throw on `this.currentMediaInfo_.hasAudio`

### CODE REVIEW

File: src/segment-loader.js [hunk 1]
```diff
@@ -1917,15 +1927,13 @@ export default class SegmentLoader extends videojs.EventTarget {
     }
 
     const segmentInfo = this.pendingSegment_;
+    const trackInfo = this.getCurrentMediaInfo_();
 
-    // no segment to append any data for or
-    // we do not have information on this specific
-    // segment yet
```
Comment (gkatsev) [inline src/segment-loader.js:1923]: is this comment no longer accurate?

File: src/segment-loader.js [hunk 2]
```diff
@@ -1905,6 +1907,14 @@ export default class SegmentLoader extends videojs.EventTarget {
     return true;
   }
 
+  getCurrentMediaInfo_(segmentInfo = this.pendingSegment_) {
+    return segmentInfo && segmentInfo.trackInfo || this.currentMediaInfo_;
```
Comment (gkatsev) [inline src/segment-loader.js:1911]: Am I reading this right that basically `currentMediaInfo_` is the trackInfo, but that there's a sync issue between what the SegmentLoader has (or doesn't have) and what the actual segmentInfo is. Does it make sense to even have `currentMediaInfo_` at all? Should we remove it and rely only on segmentInfo.trackInfo?

File: src/segment-loader.js (hunk 1 above)
Comment (brandonocasey) [reply src/segment-loader.js]: this was removed in error.

File: src/segment-loader.js (hunk 2 above)
Comment (brandonocasey) [reply src/segment-loader.js:1911]: I was wondering the same myself during the creation of this pull request but was running into issues in a few areas trying to get segmentInfo.trackInfo working across the board. I think it's likely that we should do this in the future, but for now we should stick with this. I will add a task to #1057 to think about why we need current/starting media info and wether we should just be using `segmentInfo.trackInfo`.

Comment (gkatsev) [reply src/segment-loader.js:1911]: Sounds good.

**Review by gkatsev — APPROVED** [review]

**Pull request merged**